bin/frequify.ml

```diff
@@ -0,0 +1,307 @@
+open Mtyped
+open Term
+module Env = Zzenv
+
+(** calls poirot to get AST of source_file *)
+let process meta_config_file source_file () =
+  let () = Env.load_meta meta_config_file in
+  let code = Commands.Cre.preprocess source_file () in
+  code
+
+(** returns "frequency_gen" as ('t, string) typed *)
+let replace_bool_gen_string (s : ('t, string) typed) : ('t, string) typed = 
+  s #-> (function _ -> "frequency_gen")
+
+let is_bool_gen (s : ('t, string) typed) : bool =
+  s.x = "bool_gen"
+
+(** recursively traverses through AST to find and replace bool_gen with frequency_gen *)
+let rec replace_bool_gen (t : ('t, 't term) typed) : ('t, 't term) typed = 
+  t #-> ( function
+  | CErr -> CErr
+  | CVal t -> CVal t #-> replace_bool_gen_value
+  (* thunkifies branches *)
+  | CLetE { 
+      lhs; 
+      rhs = { x = CApp { appf = { x = VVar {x = "bool_gen"; ty }; ty = ty2} ; apparg }; ty = ty3 }; 
+      body = { x = CMatch { matched ; match_cases = [
+        CMatchcase
+          { 
+            constructor = { x = "True"; ty = ty_t }; 
+            args = []; 
+            exp = exp1;
+          };
+        CMatchcase
+          {
+            constructor = { x = "False"; ty = ty_f };
+            args = [];
+            exp = exp2;
+          };
+      ]; }; ty = ty4} } ->
+        CLetE {     (* w0 = get_weight_idx *)
+          lhs = ("w0" #: ty);
+          rhs = { x = CApp {
+            appf = { x = VVar ("get_weight_idx" #: ty); ty = Nt.Ty_any};
+            apparg = {x = VConst (I 0); ty = Nt.Ty_any};
```

Comment:
This needs the additional runtime weights I guess right?

Reply:
Is it safe for me to just replace this with Const (I 1) since I always default the first one to that